core/tee/tee_svc_storage.c

```diff
@@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 
 	fops = o->pobj->fops;
 
+	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
```

Comment:
What your code do now is head->ds_size = o->info.dataSize, just overwrite the head without the update from B?